pkg/controller/datavolume-controller.go

```diff
@@ -352,20 +370,37 @@ func (c *DataVolumeController) syncHandler(key string) error {
 		return errors.Errorf(msg)
 	}
 
+	// expectations prevent us from creating multiple pods. An expectation forces
+	// us to observe a pod's creation in the cache.
 	needsSync := c.pvcExpectations.SatisfiedExpectations(key)
+
 	if !exists && needsSync {
-		newPvc, err := newPersistentVolumeClaim(dataVolume)
-		if err != nil {
-			return err
-		}
-		c.pvcExpectations.ExpectCreations(key, 1)
-		pvc, err = c.kubeclientset.CoreV1().PersistentVolumeClaims(dataVolume.Namespace).Create(newPvc)
-		if err != nil {
-			c.pvcExpectations.CreationObserved(key)
-			return err
-		}
-		if canUpdateProgress(newPvc.Annotations) {
-			go c.scheduleProgressUpdate(dataVolume.Name, dataVolume.Namespace, pvc.GetUID())
+		snapshotClassName := c.getSnapshotClassForSmartClone(dataVolume)
+		if snapshotClassName != "" {
+			klog.V(3).Infof("Smart-Clone via Snapshot is available with Volume Snapshot Class: %s", snapshotClassName)
+			newSnapshot := newSnapshot(dataVolume, snapshotClassName)
+			_, err := c.csiClientSet.SnapshotV1alpha1().VolumeSnapshots(newSnapshot.Namespace).Create(newSnapshot)
+			if err != nil {
+				return err
+			}
+			err = c.updateSmartCloneStatusPhase(cdiv1.SnapshotForSmartCloneInProgress, dataVolume)
```

Comment:
Looks like the phase is updated in the smart clone controller as well.  Can we keep it in one place?  I think it would be nice and consistent to do in the datavoume controller.  But would be more work as would have to watch snapshots.

Reply:
Later is fine